crates/network-scanner/src/mdns.rs

```diff
@@ -136,21 +109,47 @@ fn parse_fullname(fullname: &str) -> Option<(String, Option<Protocol>)> {
     // remove the trailing dot
     service_type.pop()?;
 
-    let protocol = match service_type.as_str() {
-        "_http._tcp" => Some(Protocol::Http),
-        "_https._tcp" => Some(Protocol::Https),
-        "_ssh._tcp" => Some(Protocol::Ssh),
-        "_sftp._tcp" => Some(Protocol::Sftp),
-        "_scp._tcp" => Some(Protocol::Scp),
-        "_telnet._tcp" => Some(Protocol::Telnet),
-        "_ldap._tcp" => Some(Protocol::Ldap),
-        "_ldaps._tcp" => Some(Protocol::Ldaps),
-        // https://jonathanmumm.com/tech-it/mdns-bonjour-bible-common-service-strings-for-various-vendors/
-        // OSX Screen Sharing
-        "_rfb._tcp" => Some(Protocol::Ard),
-        // Note: RDP, VNC, Wayk, and SSH-based PowerShell (SshPwsh) are not standardized service types
-        _ => None,
-    };
+    let protocol = service_type.as_str().try_into().ok();
 
     Some((device_name.to_string(), protocol))
 }
+
+impl TryFrom<&str> for ServiceType {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "_http._tcp" => Ok(ServiceType::Http),
+            "_https._tcp" => Ok(ServiceType::Https),
+            "_ssh._tcp" => Ok(ServiceType::Ssh),
+            "_sftp._tcp" => Ok(ServiceType::Sftp),
+            "_scp._tcp" => Ok(ServiceType::Scp),
+            "_telnet._tcp" => Ok(ServiceType::Telnet),
+            "_ldap._tcp" => Ok(ServiceType::Ldap),
+            "_ldaps._tcp" => Ok(ServiceType::Ldaps),
+            // https://jonathanmumm.com/tech-it/mdns-bonjour-bible-common-service-strings-for-various-vendors/
+            // OSX Screen Sharing
+            "_rfb._tcp" => Ok(ServiceType::Ard),
+            "_rdp._tcp" | "_rdp._udp" => Ok(ServiceType::Rdp),
+            _ => Err(anyhow::anyhow!("Unknown protocol: {}", value)),
```

Comment:
style: Should start with a lowercase.
```suggestion
            _ => Err(anyhow::anyhow!("unknown protocol: {}", value)),
```